internal/pkg/vault/secrets_test.go

```diff
@@ -865,3 +898,170 @@ func (caller *InMemoryMockCaller) Do(req *http.Request) (*http.Response, error)
+			expectedValues:    []string{"one", "two", "three/"},
+			expectError:       false,
+			expectedErrorType: nil,
+			expectedDoCallNum: 1,
+			caller: &InMemoryMockCaller{
+				DataList: testData[testPath],
+			},
+		},
+		{
+			name:              "No keys error",
+			path:              testPath2,
+			expectedValues:    []string{},
+			expectError:       false,
+			expectedErrorType: nil,
+			expectedDoCallNum: 1,
+			caller: &InMemoryMockCaller{
+				DataList: testData[testPath2],
+			},
+		},
+		{
+			name:              "Subpath",
+			path:              testPath3,
+			expectedValues:    nil,
+			expectError:       false,
+			expectedErrorType: nil,
+			expectedDoCallNum: 1,
+			caller: &InMemoryMockCaller{
+				DataList: testData[testPath3],
+			},
+		},
+		{
+			name:              "Get non-existent Key",
+			path:              "/one",
+			expectedValues:    nil,
+			expectError:       true,
+			expectedErrorType: pkg.NewErrPathNotFound("Does not exist"),
+			expectedDoCallNum: 1,
+			caller: &ErrorMockCaller{
+				ReturnError: false,
+				StatusCode:  404,
+				ErrorType:   pkg.NewErrPathNotFound("Does not exist"),
+			},
+		},
+		{
+			name:              "Get all Keys",
+			path:              "",
+			expectedValues:    []string{"one", "two", "three/"},
+			expectError:       false,
+			expectedErrorType: nil,
+			expectedDoCallNum: 1,
+			caller: &InMemoryMockCaller{
+				DataList: testData[testPath],
+			},
+		},
+		{
+			name:              "Handle HTTP no path error",
+			path:              testPath,
+			expectedValues:    nil,
+			expectError:       true,
+			expectedErrorType: TestConnErrorPathNotFound,
+			expectedDoCallNum: 1,
+			caller: &ErrorMockCaller{
+				ReturnError: false,
+				StatusCode:  404,
+				ErrorType:   pkg.NewErrPathNotFound("Not found"),
+			},
+		},
+		{
+			name:              "Handle non-200 HTTP response",
+			path:              testPath,
+			expectedValues:    nil,
+			expectError:       true,
+			expectedErrorType: TestConnError,
+			expectedDoCallNum: 1,
+			caller: &ErrorMockCaller{
+				ReturnError: false,
+				StatusCode:  400,
+				ErrorType:   pkg.NewErrSecretStore("Error"),
+			},
+		},
+		{
+			name:              "Get Key with unknown path",
+			path:              "/nonexistentpath",
+			expectedValues:    nil,
+			expectError:       true,
+			expectedErrorType: TestConnErrorPathNotFound,
+			expectedDoCallNum: 1,
+			caller: &InMemoryMockCaller{
+				DataList: testData[testPath2],
+			},
+		},
+		{
+			name:              "URL Error",
+			path:              "bad path for URL",
+			expectedValues:    nil,
+			expectError:       true,
+			expectedErrorType: errors.New(""),
+			caller: &InMemoryMockCaller{
+				DataList: testData[testPath2],
+			},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			cfgHTTP := types.SecretConfig{
+				Host:      "localhost",
+				Port:      8080,
+				Protocol:  "http",
+				Namespace: testNamespace,
+			}
+			ssm := Client{
```

Comment:
Use more meaningful names so code is more readable.

```suggestion
			client := Client{
```